### ISSUE TITLE
Incompatible GoogleAppMeasurement version on cordova ios build.

### ISSUE THREAD

**Issue by danieleisenhardt** [admob-plus/admob-plus#210]
`cd platforms/ios`

`pod install`

```
[!] CocoaPods could not find compatible versions for pod "GoogleAppMeasurement":
  In snapshot (Podfile.lock):
    GoogleAppMeasurement (= 7.3.0, ~> 7.0)

  In Podfile:
    Firebase/Analytics (~> 7.0.0) was resolved to 7.0.0, which depends on
      Firebase/Core (= 7.0.0) was resolved to 7.0.0, which depends on
        FirebaseAnalytics (= 7.0.0) was resolved to 7.0.0, which depends on
          GoogleAppMeasurement (= 7.0.0)

    Google-Mobile-Ads-SDK (~> 7.66) was resolved to 7.69.0, which depends on
      GoogleAppMeasurement (~> 7.0)
```
I suppose CocoaPods thinks GoogleAppMeasurement 7.0.0 is different from GoogleAppMeasurement 7.0 

I tried changing xml files in both plugins in order to make a fix, fork and pull request, but this didn't help.

**Comment by darkguy2008** [comment]
This project does not seem active. Does anybody know an alternative we can use? For iOS if possible?

**Comment by danieleisenhardt** [comment]
> This project does not seem active. Does anybody know an alternative we can use? For iOS if possible?

There's plenty of others, but they all steal your ad revenue AFAIK

**Comment by ratson** [comment]
Close as related to the old version.

**Comment by rideo-eduardmorales** [comment]
Hi, we are getting the same error using latest version "admob-plus-cordova": "1.4.0". Relevant parts (in our opinion) of our package.json:
"cordova-ios": "6.2.0",
"cordova-plugin-cocoapod-supportx": "1.7.3",
"cordova-plugin-firebasex": "11.0.3-cli"
"cordova-plugin-googleplus": "8.5.1"
"cordova-android-firebase-gradle-release": "4.0.0",
"cordova-android-play-services-gradle-release": "4.0.0",
"cordova-android-support-gradle-release": "3.0.1"

Ionic:

   Ionic CLI          : 5.4.16 (/usr/local/lib/node_modules/ionic)
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.2.4

Cordova:

   Cordova CLI       : 10.0.0
   Cordova Platforms : 6.0.0, android 9.0.0, browser, ios 6.2.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 5.0.0, (and 21 other plugins)

Utility:

   cordova-res : not installed
   native-run  : 1.3.0

System:

   Android SDK Tools : 25.2.5 (/Users/eduard/Library/Android/sdk)
   ios-deploy        : 1.11.4
   ios-sim           : 8.0.2
   NodeJS            : v10.24.0 (/usr/local/Cellar/node@10/10.24.0_1/bin/node)
   npm               : 6.14.11
   OS                : macOS Catalina
   Xcode             : Xcode 12.3 Build version 12C33


Any ideas? Thanks in advance.

**Comment by ratson** [comment]
@rideo-eduardmorales If you could setup a test repo, then I will check the problem.

**Comment by danieleisenhardt** [comment]
I'm sorry to say that I do not have the time right now. Maybe later, unless rideo-eduardmorales beats me to it.

**Comment by rideo-eduardmorales** [comment]
OK, I'll try to setup it and I'll let you know. Thanks.

**Comment by rideo-eduardmorales** [comment]
Hi @ratson, I've set up a repo:

https://github.com/team-rideosoftware/admob-plus-test

After cloning, to replicate the error follow these steps:

1. Add admob-plus plugin with sample admob App ID from Google(iOS: https://developers.google.com/ad-manager/mobile-ads-sdk/ios/quick-start, Android: https://developers.google.com/ad-manager/mobile-ads-sdk/android/quick-start)

`ionic cordova plugin add admob-plus-cordova --save --variable APP_ID_ANDROID=ca-app-pub-3940256099942544~3347511713 --variable APP_ID_IOS=ca-app-pub-3940256099942544~1458002511`

This show an error and stops there. Then, to clearly see the error you can do:

`cd platforms/ios && pod install --repo-update`

And that will show:

`Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "GoogleAppMeasurement":
  In snapshot (Podfile.lock):
    GoogleAppMeasurement (= 6.8.3)

  In Podfile:
    Firebase/Core (= 6.33.0) was resolved to 6.33.0, which depends on
      FirebaseAnalytics (= 6.8.3) was resolved to 6.8.3, which depends on
        GoogleAppMeasurement (= 6.8.3)

    Google-Mobile-Ads-SDK (~> 8.3.0) was resolved to 8.3.0, which depends on
      GoogleAppMeasurement (~> 7.0)
`

Hope this helps you identify and hopefully solve the error. Thanks in advance.

Eduard

**Comment by ratson** [comment]
Update `cordova-plugin-firebasex` to v12 will solve the issue.

**Comment by rideo-eduardmorales** [comment]
Thank you for your pull request. I did not specify that we had tried to update to v12 the firebasex plugin, but then another conflict appeared with cordova-plugin-googleplus, and that's why we discarded that option. 

So, haveing accepted that we have to update to v12, and I know that this may not be related to admob-plus, but since you understand these things better than I do, I would really appreciate if you can try this and tell me if you know a solution. If you try to add the cordova-plugin-googleplus to the test repository, you will get the following error:

```
Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "GoogleUtilities/NSData+zlib":
  In snapshot (Podfile.lock):
    GoogleUtilities/NSData+zlib (= 7.3.1, ~> 7.0)

  In Podfile:
    Firebase/RemoteConfig (= 7.8.1) was resolved to 7.8.1, which depends on
      FirebaseRemoteConfig (~> 7.8.0) was resolved to 7.8.0, which depends on
        GoogleUtilities/NSData+zlib (~> 7.0)

    GoogleUtilities (~> 7.2.2) was resolved to 7.2.2, which depends on
      GoogleUtilities/NSData+zlib (= 7.2.2)

```

We have tried adding the cordova-plugin-cocoapod-supportx and specify the version with:
```
<preference name="pods_use_frameworks" value="true" />      
<pod name="GoogleUtilities" spec="7.3.1" />
```
but the problem is still there. Any ideas?

**Comment by rideo-eduardmorales** [comment]
Hi @ratson, forget my last message, we decided to get rid of cordova-plugin-googleplus. Now, we have admob-plus-cordova added to our ionic 3 test project, it builds correctly, and we followed the plugin documentation to show an interstitial ad, but the following line:
```
await this.admob.start(); // throws error "undefined is not an object (evaluating 'this.admob.start')"
```
throws this error in the console:
```
Cannot read property 'start' of undefined. TypeError: Cannot read property 'start' of undefined
```

Could you please take a look at the test repo and tell us if there is a fix we could apply? Since we are using ionic 3 for other project requirements, we don't know if there could be an unmet dependency that throws this error, and if it is possible to find a workaround.

Thanks in advance.
